vscode-extension/src/aiConfigEditorManager.ts

```diff
@@ -0,0 +1,83 @@
+import vscode from "vscode";
+import { ServerInfo } from "./util";
+
+export class AIConfigEditorState {
+  constructor(
+    public document: vscode.TextDocument,
+    public readonly webviewPanel: vscode.WebviewPanel,
+    public editorServer: ServerInfo | null,
+    private readonly manager: AIConfigEditorManager
+  ) {
+    // Listen to when the panel's view state changes and update the active editor
+    this.webviewPanel.onDidChangeViewState((e) => {
+      if (e.webviewPanel.active) {
+        this.manager.setActiveEditor(this);
+      } else if (this.manager.getActiveEditor() === this) {
+        this.manager.setActiveEditor(null);
+      }
+    });
+
+    this.webviewPanel.onDidDispose(() => {
+      // If the webview panel has been closed, remove it from the manager as well
+      this.manager.removeEditor(this);
+    });
+  }
+}
+
+export class AIConfigEditorManager {
+  private activeEditor: AIConfigEditorState | null;
+  private activeEditorUri: string;
+  private editorsByUri: Map<string, AIConfigEditorState> = new Map();
+
+  constructor() {
+    this.activeEditor = null;
+    this.activeEditorUri = null;
+  }
+
+  setActiveEditor(editor: AIConfigEditorState | null) {
+    this.activeEditor = editor;
+    this.activeEditorUri = editor?.document.uri.toString();
```

Comment:
Can this variable become outdated if changing the filename while it's open? Just wondering if we should be getting the uri directly / at time-of-need to ensure it's always up-to-date instead of setting it in a variable